src/main/java/org/opensearch/knn/index/codec/KNN80Codec/KNN80DocValuesConsumer.java

```diff
@@ -354,4 +348,21 @@ private boolean isChecksumValid(long value) {
         // https://github.com/apache/lucene/blob/branch_9_0/lucene/core/src/java/org/apache/lucene/codecs/CodecUtil.java#L644-L647
         return (value & CRC32_CHECKSUM_SANITY) != 0;
     }
+
+    private boolean isBinaryField(FieldInfo field) {
+        if (field.attributes().containsKey(MODEL_ID)) {
+            Model model = ModelCache.getInstance().get(field.attributes().get(MODEL_ID));
```

Comment:
Model can be quite big because it contains the binary blob. Can we read from model metadata instead via ModelDao?

Reply:
Updated code to use model only once in the addKNNBinaryField function